specific/cmo_stock/models/stock.py

```diff
@@ -0,0 +1,132 @@
+# -*- coding: utf-8 -*-
+from openerp import models, fields, api
+from openerp.tools.translate import _
+
+
+class StockPicking(models.Model):
+    _inherit = 'stock.picking'
+
+    partner_id = fields.Many2one(
+        'res.partner',
+        default=lambda self: self._get_partner_id(),
```

Comment:
You can use **self.env.user** to get the user. And so this case it is as easy as,
`default=lambda self: self.env.user.partner`
Note: no need .id